### PR TITLE
LPS-201631 StackOverflow in Web Content

### PR DESCRIPTION
**Pull request by yuhai**
@dantewang, please see the [ticket's](https://liferay.atlassian.net/browse/LPS-201631) description.

### CODE REVIEW

**Comment by yuhai** [comment]
ci:test:relevant

**Comment by Kvale1733** [comment]
@yuhai No regressions found. Testing passed following ticket steps.

File: modules/apps/journal/journal-service/src/main/java/com/liferay/journal/internal/transformer/JournalTransformer.java [hunk 1]
```diff
@@ -1006,5 +1025,10 @@ private String _getTemplatesPath(
 		JournalTransformer.class.getName() + ".XmlAfterListener");
 	private static final Log _logXmlBeforeListener = LogFactoryUtil.getLog(
 		JournalTransformer.class.getName() + ".XmlBeforeListener");
+	private static final ThreadLocal<Set<String>>
+		_renderedArticleIdsThreadLocal = new CentralizedThreadLocal(
```
Comment (dantewang) [inline modules/apps/journal/journal-service/src/main/java/com/liferay/journal/internal/transformer/JournalTransformer.java:1029]: @yuhai Raw type here, you need to add `<>`.

File: modules/apps/journal/journal-test/src/testIntegration/java/com/liferay/journal/transformer/test/JournalTransformerTest.java [hunk 2]
```diff
@@ -237,6 +267,135 @@ public void testLocaleTransformerListenerNestedFieldWithNoTranslation()
...
+
+		Configuration configuration = null;
+
+		Dictionary<String, Object> originalProperties = null;
+
+		try (LogCapture logCapture = LoggerTestUtil.configureLog4JLogger(
+				"com.liferay.journal.internal.transformer.JournalTransformer",
+				LoggerTestUtil.WARN)) {
+
+			configuration = _configurationAdmin.getConfiguration(
```
Comment (dantewang) [inline modules/apps/journal/journal-test/src/testIntegration/java/com/liferay/journal/transformer/test/JournalTransformerTest.java:288]: @yuhai Can you use `ConfigurationTestUtil`?

File: modules/apps/journal/journal-service/src/main/java/com/liferay/journal/internal/transformer/JournalTransformer.java [hunk 3]
```diff
@@ -344,6 +361,8 @@ else if (exception instanceof TransformException) {
 				httpServletRequest.setAttribute(
 					PortletRequest.LIFECYCLE_PHASE, originalLifecyclePhase);
 			}
+
+			renderedArticleIds.remove(articleId);
```
Comment (dantewang) [inline modules/apps/journal/journal-service/src/main/java/com/liferay/journal/internal/transformer/JournalTransformer.java:365]: @yuhai As a post-execution clean-up, this is not guaranteed to run. The try-finally scope is pretty small compared to the length of this method. Please consider this.

**Review by dantewang — CHANGES_REQUESTED** [review]

**Comment by dantewang** [comment]
Plz resend.

**Pull request closed without merging**